apps/rush-lib/src/logic/taskRunner/TaskRunner.ts

```diff
@@ -177,7 +184,17 @@ export class TaskRunner {
 
     this._terminal.writeStdoutLine(`Executing a maximum of ${this._parallelism} simultaneous processes...`);
 
-    await this._startAvailableTasksAsync();
+    if (this._tracer) {
+      this._traceQueueStatus();
+    }
+
+    const maxParallelism: number = Math.min(this._buildQueue.length, this._parallelism);
+    const workers: Promise<void>[] = [];
+    for (let i: number = 0; i < maxParallelism; i++) {
+      workers[i] = this._taskWorker(i + 1);
+    }
+
+    await Promise.all(workers);
```

Comment:
Will this wait for all currently executing tasks to finish before queueing more tasks? That seems wrong.

Reply:
Each worker consumes tasks from the queue one at a time until the task queue is empty. If the task queue is not empty but not tasks are available, the workers suspend, and after a task has been completed, the worker hands off available tasks to suspended workers until there are no more available tasks or no more suspended workers.